### ISSUE TITLE
Is there any way to save state from a browser and restore it later?

### ISSUE THREAD

**Issue by bitcoinmeetups** [floooh/chips-test#25]
Hi,

I'm BM, a very nice and polite guy.

Is there any way to save state from a browser and restore it later?

I would like to manipulate the states and then embed them so other people can watch a particular state and interact with it. Browser only, I don't want to install anything. Embeddable.

**Comment by floooh** [comment]
The new snapshot feature doesn't support this, because it's very brittle (e.g. it's not compatible between emulator version updates) and is only meant for debugging.

Some emulators support standard snapshot formats though (for instance C64 support PRG and CPC supports SNA).

**Comment by floooh** [comment]
Nope.

**Comment by floooh** [comment]
Currently there isn't any 'infrastructure' to get such standard snapshot files out of the emulator. Can't you just use a different emulator to generate the SNA? The chips emulators are mostly just for 'consumption' of ROMs and snapshots.

**Comment by diejuse** [comment]
> Currently there isn't any 'infrastructure' to get such standard snapshot files out of the emulator. Can't you just use a different emulator to generate the SNA? The chips emulators are mostly just for 'consumption' of ROMs and snapshots.
> 
> (although such a feature _would_ make sense in context of https://marketplace.visualstudio.com/items?itemName=floooh.vscode-kcide, but it will be quite a while until I get back to emulator work (earliest is probably next Christmas vacation).

Yes , I can use a different emulator to generate the SNA But I think your emulator would be perfect if it could produce and load saved states in files, whatever format. That way I could continue the games on any of my devices, which are many 😅
Thank you for your attention and for your excellent work.
